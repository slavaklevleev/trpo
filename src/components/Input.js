import React from "react";
import TextareaAutosize from "react-textarea-autosize";

class Input extends React.Component {
  constructor(props) {
    super(props);
    this.handleTextChange = this.handleTextChange.bind(this);
  }

  handleTextChange(e) {
    this.props.onTextChange(e.target.value);
  }

  render() {
    return (
      <div className="input">
        <h2>Введите текст:</h2>
        <TextareaAutosize
          minRows={15}
          value={this.props.text}
          onChange={this.handleTextChange}
        />
      </div>
    );
  }
}

export default Input;
export { Input };