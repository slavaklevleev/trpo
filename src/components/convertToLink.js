const convertToLink = (text) => {
  var regexp = /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_.~#?&//=]*)/g;
  return text
    .replace(regexp, "<a href='https://$&'>https://$&</a>")
    .split("\n")
    .join("<br/>");
};

export default convertToLink;