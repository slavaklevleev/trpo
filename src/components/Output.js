import React from "react";
import convertToLink from "./convertToLink"

class Output extends React.Component {
  render() {
    return (
      <div className="output">
        <h2>Ваш текст:</h2>
        <p
          dangerouslySetInnerHTML={{
            __html: convertToLink(this.props.text),
          }}
        />          
      </div>
    );
  }
}

export { Output };