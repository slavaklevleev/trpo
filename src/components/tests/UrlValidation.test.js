import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { create } from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { Output } from "./../Output";
import convertToLink from "./../convertToLink";
import Input from "../Input";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Converting to link works correctly", () => {
  test("Function is defined", () => {
    const text = "vk.com";
    expect(convertToLink).toBeDefined();
  });

  test("Convert link to link", () => {
    expect(convertToLink("vk.com")).toEqual(
      "<a href='https://vk.com'>https://vk.com</a>"
    );
    expect(convertToLink("www.ya.ru")).toEqual(
      "<a href='https://www.ya.ru'>https://www.ya.ru</a>"
    );
    expect(convertToLink("google.com")).toEqual(
      "<a href='https://google.com'>https://google.com</a>"
    );
    expect(convertToLink("foo.com/blah_blah/")).toEqual(
      "<a href='https://foo.com/blah_blah/'>https://foo.com/blah_blah/</a>"
    );
    expect(convertToLink("userid@example.com:8080")).toEqual(
      "<a href='https://userid@example.com:8080'>https://userid@example.com:8080</a>"
    );
  });

  test("Do not convert reqular text to link", () => {
    expect(convertToLink("word")).toEqual("word");
    expect(convertToLink("http://")).toEqual("http://");
    expect(convertToLink("http://?")).toEqual("http://?");
    expect(convertToLink("http://##/")).toEqual("http://##/");
    expect(convertToLink("h://test")).toEqual("h://test");
  });
});

describe("Output testing", () => {
  test("Render Output with emty props", () => {
    act(() => {
      render(<Output text="" />, container);
    });
    expect(container.textContent).toBe("Ваш текст:");
  });

  test("Render Output with some text", () => {
    act(() => {
      render(<Output text="some text" />, container);
    });
    expect(container.textContent).toBe("Ваш текст:some text");
  });

  test("Render Output with some text and link", () => {
    act(() => {
      render(<Output text="some text example.com" />, container);
    });
    expect(container.textContent).toBe(
      "Ваш текст:some text https://example.com"
    );
  });
});

describe("Input testing", () => {
  test("Matches the snapshot", () => {
    const component = create(<Input />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
