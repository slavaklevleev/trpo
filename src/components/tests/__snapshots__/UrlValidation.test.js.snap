// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input testing Matches the snapshot 1`] = `
<div
  className="input"
>
  <h2>
    Введите текст:
  </h2>
  <textarea
    onChange={[Function]}
    style={
      Object {
        "height": 0,
      }
    }
  />
</div>
`;
